src/rust/engine/src/scheduler.rs

```diff
@@ -331,6 +336,23 @@ impl Scheduler {
     })
   }
 
+  async fn wait_for_tail_tasks(tasks: Vec<BoxFuture<'static, ()>>) {
+    if !tasks.is_empty() {
+      log::trace!("waiting for {} tail tasks to complete", tasks.len());
```

Comment:
Probably debug?

Reply:
Would it be useful to print out "names" of the tasks that were delayed? (And then I'd add a newtype to carry the task future and its name.)

Reply:
Very, yea.